### DISCUSSION TITLE
idx name is changed for EV1 model after  loading in ANDES

### DISCUSSION THREAD

**Discussion by PandaRP01** [CURENT/andes#482] [Q&A] [answered]
Hello Dr. Cui

I am faced a strange issues while simulating power system network with EV1 model.

While preparing ANDES input file, I have assigned idx number for models what I used.

In  my model, 31  battery (ESD1), 37 solar (PVD1) AND 30 Electric Vechile (EV1).
However, while simulating in ANDES, when I run `ssPV.EV1.cache.df `command, I observed, that the idx number for EV1 is changed. 

In the ANDES input model excel file I defined it as 188 , 189,.. for EV1 models but after running in ANDES,  the idx number are displayed as  EV1_69, EV1_70.... Also ANDES commands accept these idx number, i.e. EV1_69, EV1_70,....
However, for ESD1 and PVD1, the idx numbers do not change . it remains same as O defined in input model.
Could you please address this confusion.

Regards
Rakesh Panda

**Comment by cuihantao** [comment, marked as answer]
It’s possible that you have clashing idxes, and ANDES automatically renamed these idxes to allow simulations to run.

 

EV1, ESD1 and PVD1 belong to the same group, so all their idxes must be unique.

 

From: PandaRP01 ***@***.***> 
Sent: Wednesday, September 13, 2023 2:57 PM
To: CURENT/andes ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [CURENT/andes] idx name is changed for EV1 model after loading in ANDES (Discussion #482)

 

Hello Dr. Cui

I am faced a strange issues while simulating power system network with EV1 model.

While preparing ANDES input file, I have assigned idx number for models what I used.

In my model, 31 battery (ESD1), 37 solar (PVD1) AND 30 Electric Vechile (EV1).
However, while simulating in ANDES, when I run ssPV.EV1.cache.df command, I observed, that the idx number for EV1 is changed.

In the ANDES input model excel file I defined it as 188 , 189,.. for EV1 models but after running in ANDES, the idx number are displayed as EV1_69, EV1_70.... Also ANDES commands accept these idx number, i.e. EV1_69, EV1_70,....
However, for ESD1 and PVD1, the idx numbers do not change . it remains same as O defined in input model.
Could you please address this confusion.

Regards
Rakesh Panda

**Reply by PandaRP01** [reply]
Hello Dr. Cui

Thank you so much for addressing my issues.
As you have highlighted, I had clashes of ide numbers between `ESD1` and `EV1`.

If possible,  It could be good if you can add a flashing message as warning or error before ANDES generate his own idx (while ANDES loading), when this conflict are present in the input model. That could be very helpful to know the problems.

Regards
Rakesh Panda

**Reply by cuihantao** [reply]
Check if there are already warnings? See https://github.com/CURENT/andes/blob/620373198bfcbfa8825020d3edc38ae195bea3fe/andes/models/group.py#L376-L383

**Reply by PandaRP01** [reply]
Hello Dr. Cui
Yes, this warning I had. Unable to debug.

Thank you so much
Regards
Rakesh
